server/server.go

```diff
@@ -119,9 +119,20 @@ func (s *Server) startHttpServer() error {
 		ReadHeaderTimeout: 30 * time.Second,
 	}
 
-	s.logger.Info("Starting standalone HTTP server", zap.String("address", s.cfg.HTTPListener))
-	if err := httpServer.ListenAndServe(); err != nil && err != http.ErrServerClosed {
-		return fmt.Errorf("HTTP server failed: %w", err)
+	listener, err := net.Listen("tcp", s.cfg.HTTPListener)
+	if err != nil {
+		return fmt.Errorf("failed to create HTTP listener: %w", err)
 	}
```

Comment:
is this needed? we already hv `httpServer` created above that sets `s.cfg.HTTPListener`

do you think simply wrapping the existing `ListenAndServe` code in a goroutine will be enough?

Reply:
the listener way ensures the server is available before proceeding

Reply:
what do you mean by "server is available"? do you mean this can detect if the port is used?

but that's no different than calling `ListenAndServe` which calls `net.Listen` in it. and `ListenAndServe` make our codebase simpler to read

```
func (srv *Server) ListenAndServe() error {
	if srv.shuttingDown() {
		return ErrServerClosed
	}
	addr := srv.Addr
	if addr == "" {
		addr = ":http"
	}
	ln, err := net.Listen("tcp", addr)
	if err != nil {
		return err
	}
	return srv.Serve(ln)
}
```

Reply:
> ListenAndServe would fail silently in goroutine

I think that's a different issue we need to fix. ideally we shouldn't have anything fail silently in goroutine b/c`Serve` can also fail

Reply:
imo this is still needed as there are some other cases when listening to a port will fail, e.g., there is another process listening at the same port.

Reply:
Right but my point is ListenAndServe also does the same check if you see the implementation.

I think a more important question is how to avoid silent failure inside the go routine

Reply:
offline discussion, add TODO for silent failure in goroutine where server is down but main thread keeps running.

Reply:
yes the actual issue is `Serve` can silently fail. meaning the docker container might be running but the services are actually down. this should be caught and fixed. probably using a channel and crash the container instead of fail silently inside the go routine